### ISSUE TITLE
dev build improvement

### ISSUE THREAD

**Issue by ocehugo** [aodn/imos-toolbox#564]
The building of the GUI is very monolithic and not very flexible.
This issue is to track the features of an alternative way to build the binaries.

Currently, it's a bit slow, it requires one to double clone the repository, depends on hardcoded paths and relative paths. A Matlab function is also used to build the deployment, which is not completely necessary ( we only need to call ant/mcc). If we can build without the Matlab executable dependency, then we can easily auto-deploy through Jenkins without licenses (the matlab runtime is "free").
Cross-Compilation support would also be a good thing. This could save a lot of time.

Improvements here include:

- [x] A flexible build to a specific folder
- [x] Default build to current master/tag
- [x] Allow specification of the toolbox directory
- [x] no dependency on relative paths

**Comment by petejan** [comment]
Could you get rid of the geomag c code, by using a matlab implementation, this would remove some of the machine dependency in the current code base.

**Comment by ocehugo** [comment]
@petejan, I had a quick look and appears that geomag70.c is a relatively big code. Unless there is a simple one-to-one match in MatLab, I don't think we can get rid of it anytime soon. I'll have to investigate more. Let me know if this has an alternative toolbox or set of routines.

**Comment by petejan** [comment]
There is matlab code around for it from memory needs some tweaking to make it work with the current model file

[geomag.m](http://mooring.ucsd.edu/software/matlab/doc/toolbox/geo/geomag.html)

**Comment by sspagnol** [comment]
I have used [International Geomagnetic Reference Field (IGRF) Model](https://au.mathworks.com/matlabcentral/fileexchange/34388-international-geomagnetic-reference-field-igrf-model) code from Mathwork fileexchange in some other code. Need to check equivalency, and have area in imos-toolbox to store coeffs file (updated every 5 years). Haven't read up if you store a coeffs file per 5 year block, or is the one coeff file applicable for all dates.

**Comment by ocehugo** [comment]
I'm keen to remove c dependencies if there are a robust package and proper tests to avoid breaking stuff. I'm not too keen on fileexchange since the updates are a hassle and there is usually no tests.

My limited search indicated that IGRF-12 should be used. The Aerospace toolbox may also have some equivalent functions.

by the way, I looked into the mcc compiler options, and I had to drop requirements:

mcc is part of matlab not the matlab runtime. Hence, we cannot easily obtain/install/send/etc.

cross-compilation is not supported out of the box. There is only support for the current architecture, mostly because there are no cross-architecture libraries installed (say windows libs in the Linux install package). The mcc is also not a general-purpose compiler - I was expecting too much.
